File: comparison/go.rb
# right answer:
# 0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181, 6765, 10946, 17711, 28657, 46368, 75025, 121393, 196418, 317811, 514229

def calcFibN(num)
     fib_array = [0,1]
     (num-2).times do |i|
         fib_array << (fib_array[i] + fib_array[i+1])
     end
     #fib_array.join(", ")
end
30_00_000.times {
  calcFibN(30)
}

File: comparison/go.java
import java.util.ArrayList;
import java.util.Arrays;

class go {


  static void calcFibN(int num) {
     ArrayList<Integer> fib_array = new ArrayList<Integer>(Arrays.asList(new Integer(0), new Integer(1)));
    for(int i = 0; i < num - 2; i++) {
      //fib_array << (fib_array[i] + fib_array[i+1])
      fib_array.add(fib_array.get(i) + fib_array.get(i+1));
     }
    //System.out.println(fib_array);
  }

  public static void main(String[] args) {
    int a = 1;
    int b = a ^ a;
    
    for(int i = 0; i < 3000000; i++) {
      calcFibN(30);
    }
  }

}